src/main/java/ch/uzh/soprafs22/groupmatcher/service/MatcherService.java

```diff
@@ -92,4 +91,42 @@ public void createTeams(Long matcherId) {
                 .forEach(teamEntry -> formTeam(matcher, teamEntry));
     }
 
+    public double[][] createAnswerMatrixAll(Long matcherId) {
+
+        double[][] studentAnswerMatrixSingle = createAnswerMatrixByQuestionType(matcherId, QuestionType.SINGLE_CHOICE);
+        double[][] studentAnswerMatrixMulti = createAnswerMatrixByQuestionType(matcherId, QuestionType.MULTIPLE_CHOICE);
+        double[][] studentAnswerMatrixAll = new double[studentAnswerMatrixSingle.length][];
+
+        for (int i = 0; i<studentAnswerMatrixSingle.length; i++) {
+            studentAnswerMatrixAll[i] = ArrayUtils.addAll(studentAnswerMatrixSingle[i], studentAnswerMatrixMulti[i]);
+        }
+        return studentAnswerMatrixAll;
+    }
+
+    public double[][] createAnswerMatrixByQuestionType(Long matcherId, QuestionType questionType) {
+
+        List<Long> studentIdListOrdered = studentRepository.getAllStudentsIdByMatcherId(matcherId);
+        int studentNumbers = studentIdListOrdered.size();
+
+        // stdIdListOrdered Q1-A1   Q1-A2   Q1-A3   Q2-A1   Q2-A2   Q2-A3   Q2-A4
+        // student1(0l)     1       0       1       0       1       0       1
+        // student2(1l)     0       0       1       0       1       0       0
+
+        List<Answer> answerList = answerRepository.findByQuestion_Matcher_IdAndQuestion_QuestionTypeOrderByIdAsc(matcherId,questionType);
+
+        double[] temp = new double[answerList.size()];
+        double[][] studentAnswerMatrix = new double[studentNumbers][answerList.size()];
+
+        for (int i = 0; i < studentIdListOrdered.size(); i++){
+            Optional<Student> student = studentRepository.findById(studentIdListOrdered.get(i));
+            for (int j = 0; j < answerList.size(); j++) {
+                if (student.isPresent()){
+                    temp[j] = studentRepository.existsByIdAndSelectedAnswers_Id(student.get().getId()
+                            ,answerList.get(j).getId()) ? 1 : 0;
```

Comment:
It would be better to set a value that is equal to the number of possible answers to the question instead of 1 when a student selected a particular answer. The idea here is that we want to make selecting the same answer out of 8 possible answers more "meaningful" than selected the same answer out of 3 possible answers (for example). To achieve that we add a weight to the student's selection that reflects the number of possible answers.

With your current code you should be able to get the value with answerList.get(j).getQuestion().getAnswers().size() (thought like mentioned above you can reduce the complexity of the code in general)